UtopiaCity/wwwroot/js/CityAdministration/Edit.js

```diff
@@ -0,0 +1,140 @@
+﻿window.onload = function () {
+    var id = window.location.search?.substring(4);
+    const form = document.forms["accountForm"];
+    form.elements["id"].value = id;
+    if (id != 0)
+        GetAccount(id);
+    else reset();
+}
+
+//get account by id
+async function GetAccount(id) {
+    const response = await fetch("/ResidentAccountApi/" + id, {
+        method: "GET",
+        headers: { "Accept": "application/json" }
+    });
+    if (response.ok === true) {
+        const account = await response.json();
+        const form = document.forms["accountForm"];
+        form.elements["id"].value = account.id;
+        form.elements["firstName"].value = account.firstName;
+        form.elements["familyName"].value = account.familyName;
+        form.elements["birthDate"].value = String(account.birthDate).substring(0, 10);
+        form.elements["gender"].value = account.gender;
+        form.elements["marriageId"].value = account.marriageId;
+        form.elements["birthPlace"].value = account.birthPlace;
+        form.elements["registrationAddress"].value = account.registrationAddress;
+        form.elements["medicalRecords"].value = account.medicalRecords;
+        form.elements["property"].value = account.property;
+        form.elements["motorTransport"].value = account.motorTransport;
+        form.elements["email"].value = account.email;
+        form.elements["phone"].value = account.phone;
+    }
+}
+
+// send form
+document.forms["accountForm"].addEventListener("submit", e => {
+    e.preventDefault();
+    const form = document.forms["accountForm"];
+    const id = form.elements["id"].value;
+    const firstName = form.elements["firstName"].value;
+    const familyName = form.elements["familyName"].value;
+    const birthDate = form.elements["birthDate"].value;
+    const gender = form.elements["gender"].value == "0" ? 0 : 1;
+    const marriageId = form.elements["marriageId"].value == "" ? null : form.elements["marriageId"].value;
+    const birthPlace = form.elements["birthPlace"].value;
+    const registrationAddress = form.elements["registrationAddress"].value;
+    const medicalRecords = form.elements["medicalRecords"].value;
+    const property = form.elements["property"].value;
+    const motorTransport = form.elements["motorTransport"].value;
+    const email = form.elements["email"].value;
+    const phone = form.elements["phone"].value;
+    if (id == 0)
```

Comment:
maybe it makes sense to make an object of some sort? Yes you have parsed the response, "validated" it, but 10+ parameters is too much. If only JS had strongly typed types :)

Reply:
Yes, definitely it should be done another way. I will look for a solution.

Reply:
Refactored methods with too many parameters, could you please check it?